### ISSUE TITLE
Unable to integrate solidity-coverage to a framework. Receiving ''no such file or directory, open './allFiredEvents' error

### ISSUE THREAD

**Issue by DimitarSD** [sc-forks/solidity-coverage#276]
We're developing a tool called etherlime, which is basically an ethereum development and deployment framework based on ethers.js. We want to integrate solidity-coverage in it. However, we always get this error:

```
Event trace could not be read.
Error: ENOENT: no such file or directory, open './allFiredEvents'
Exiting without generating coverage...
```
In order to run the solidity-coverage I have created the following function

```js
const runWithCoverage = async () => {
	var accountsData = ''
 	accounts.forEach(account => {
		let accountData = `--account "${account.secretKey},${account.balance.replace('0x', '')}" `;
		accountsData += accountData;
	});
	
	config["testrpcOptions"] = `${accountsData}`;
	const app = new App(config);
	app.generateCoverageEnvironment();
	app.instrumentTarget();
	app.launchTestrpc()
		.then(() => {
			app.runTestCommand();
			app.generateReport();
		})
		.catch(err => log(err));
}
```

The ``config`` variable contains the following object:

```json
{
    "compileCommand": "etherlime compile --runs=999",
    "testCommand": "etherlime test",
    "copyPackages": ["zeppelin-solidity"],
    "port": 8545,
    "buildDirPath": "/build"
}
```

When the code is executed it uses by default ``testrpc-sc`` which comes from solidity-coverage. However, I keep getting the above error. Any ideas or advice will be very helpful.

Currently, the case can be tested using the ``etherlime`` with the ``etherlime-coverage`` branch [here](https://github.com/LimeChain/etherlime/tree/etherlime-coverage). There is a [test project](https://github.com/DimitarSD/etherlime-example-test) which can be used too.

Is it possible to integrate solidity-coverage in a framework like ours? The compilation process of the smart contracts in ``etherlime`` is based on ``truffle compile`` and the generated json files from our framework are the same.

**Comment by cgewecke** [comment]
@DimitarSD Hi. Have started looking at this but am having a difficult time. As a baseline: 
```
$ git clone git clone https://github.com/LimeChain/etherlime.git # <-- master
$ cd etherlime
$ npm install
$ npm test 
```
A bunch of tests relating to ganache fail. 

```
$ npm i -g https://github.com/LimeChain/etherlime.git#etherlime-coverage
$  cd etherlime-example-test
$ etherlime test
```
Complains that it can't find `etherlime` (from the `require` statement in the LinkedList test). So then I also did:

```
$ npm init
$ npm install --save-dev https://github.com/LimeChain/etherlime.git#etherlime-coverage
```
and got a little further ... but I'm getting the same kind of errors I see in the etherlime repo tests - `Error: invalid json response`.

Would it be possible create a reproduction case for this issue that works out of the box? Ideally a sample project with all the necessary dependencies installed and some npm scripts that result in the error you're seeing above? 

Would really like to get this working - from your issue description it seems like it's probably something small that can be easily fixed.

**Comment by DimitarSD** [comment]
In order to use the etherlime commands, you should do ``npm link``.

I wrote a simple shell script below which will do everything by itself. It will use its current directory as the start point. 

```shell
red=`tput setaf 1`
reset=`tput sgr0`

echo "${red}Creating etherlime testing folder...${reset}"

mkdir etherlime-testing
cd etherlime-testing

echo "${red}Cloning the etherlime repo...${reset}"

git clone -b etherlime-coverage https://github.com/LimeChain/etherlime.git
cd etherlime

sudo npm install

echo "${red}Linking the etherlime framework...${reset}"
sudo npm link

cd ..

echo "${red}Cloning the etherlime-example-test project...${reset}"
git clone https://github.com/DimitarSD/etherlime-example-test.git
cd etherlime-example-test

echo "${red}Installing npm modules${reset}"
npm install
npm install --save zeppelin-solidity@1.5.0
npm install --save etherlime

echo "${red}Running etherlime coverage command...${reset}"
etherlime coverage
```

**Comment by cgewecke** [comment]
@DimitarSD Thanks so much, will check this out..

**Comment by cgewecke** [comment]
@DimitarSD Did you get this working? Saw a PR merge over there.

**Comment by Perseverance** [comment]
@cgewecke Yes, It is working now. I rewrote the unit tests we were using and suddenly all started working. Etherlime now has an integrated solidity-coverage inside via the `etherlime coverage` command ;)

<img width="954" alt="screen shot 2018-08-18 at 22 20 17" src="https://user-images.githubusercontent.com/5130509/44312137-18535280-a3fc-11e8-9729-2a414d4bb882.png">
